website/docs/connect-data/reference/open-ai.md

```diff
@@ -0,0 +1,222 @@
+
+* **user**: Input provided by the user to instruct or guide the model. For example, if you are using an Input widget to enter the prompt, you can use `{{Input1.text}}`.
+
+* **system**: The system message serves as a means to provide additional context, set guidelines, or convey the overall objective of the task. It helps shape the behavior of the model's responses. For example, you can use the system message to give personality to the responses or add task-specific instructions. For example, you can set system as:
+
+<dd>
+
+```html
+ "You are a technical support assistant. Provide clear and detailed solutions to user queries related to software issues. If the user mentions a bug, ask for additional details to troubleshoot effectively."
+```
+ For more information on system and user message refer the [OpenAI guide](https://platform.openai.com/docs/guides/prompt-engineering/tactic-use-delimiters-to-clearly-indicate-distinct-parts-of-the-input).
+
+</dd>
+
+
+For more information refer to the [OpenAI documentation.](https://platform.openai.com/docs/api-reference/chat/object)
+
+
+
+  <figure>
+  <img src="/img/open-ai-chat.gif" style= {{width:"810px", height:"auto"}} alt="Server-side Filtering on Table"/>
+  <figcaption align = "center"><i>OpenAI | Chat command</i></figcaption>
+</figure>
+
+
+
+
+</dd>
+
+
+### Embeddings
+
+The Embeddings command creates a vector representation of a given input, making it easier to perform tasks like semantic search or text similarity.
+
+  <figure>
+  <img src="/img/open-ai-embed.png" style= {{width:"720px", height:"auto"}} alt="Server-side Filtering on Table"/>
+  <figcaption align = "center"><i>OpenAI | Embeddings command</i></figcaption>
+</figure>
+
+#### Models
+
+<dd>
+
+Models are pre-trained language models provided by OpenAI. You can select from the available list of models, like `text-embedding-ada-002`.
+
+</dd>
+
+#### Input
+
+<dd>
+
+Input text to embed, encoded as a string or array of tokens. To embed multiple inputs in a single request, pass an array of strings or array of token arrays. You can add text information using either the Rich Text Editor or Filepicker widget, and store it in a vector database for subsequent similarity searches.
+
+</dd>
+
+
+#### Encoding format
+
+<dd>
+
+The encoding format determines how the embeddings are returned.
+
+*Options:*
+
+* **Float**: Returns the embeddings in float format.
+* **Base64**: Returns the embeddings encoded in Base64 format.
+
+</dd>
+
+
+You cannot directly bind the output to a widget, as this vector cannot be consumed on its own. Instead, you can store it in a specialized database called a Vector DB. Later, you can leverage these vectors for semantic search operations to retrieve related content based on the meaning of a query. For more information on embedding refer to the [OpenAI documentation.](https://platform.openai.com/docs/api-reference/embeddings/object)
```

Comment:
I think we need to rephrase this.
> You cannot directly bind the output to a widget

The users can definitely bind it to a widget if they wish to but there is no point in doing so. This phrase sends a message that its is not possible.

I asked ChatGPT to rephrase this. 
```suggestion
While it is technically possible to bind the output of the embeddings API to a widget, doing so offers little practical value, as the vector representation is not meaningful for direct display in the UI. Instead, you can store it in a specialized database called a Vector DB. Later, you can leverage these vectors for semantic search operations to retrieve related content based on the meaning of a query. For more information on embedding refer to the [OpenAI documentation.](https://platform.openai.com/docs/api-reference/embeddings/object)
```